s3torchconnector/src/s3torchconnector/_s3client/s3client_config.py

```diff
@@ -0,0 +1,23 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class S3ClientConfig:
+    """A dataclass exposing configurable parameters for the S3 client.
+    Returns a config wrapper object.
```

Comment:
A config wrapper object? I'd probably remove this line as it should be fairly obvious that instantiating a class gives you an instance of that class